sky/shell/tracing_controller.cc

```diff
@@ -33,7 +108,22 @@ void TracingController::StartTracing() {
   StartBaseTracing();
 }
 
-void TracingController::StopTracing(const base::FilePath& path) {
+void TracingController::StopTracing(const base::FilePath& path,
+                                    bool terminateLoopWhenDone) {
+  auto runner = base::MessageLoop::current()->task_runner();
+  runner->PostTask(FROM_HERE, base::Bind(&TracingController::StopTracingAsync,
+                                         weak_factory_.GetWeakPtr(), path,
```

Comment:
There is no lock in Dart_RegisterRootServiceRequestCallback. But this cannot be called from a random background thread after the weak factory is destroyed because we first remove the service registration on the destruction of the tracing controller.